opalog/policy_test.go

```diff
@@ -0,0 +1,292 @@
+// Copyright 2016 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package opalog
+
+import "testing"
+
+func TestPackageEquals(t *testing.T) {
+	pkg1 := &Package{Path: RefTerm(VarTerm("foo"), StringTerm("bar"), StringTerm("baz")).Value.(Ref)}
+	pkg2 := &Package{Path: RefTerm(VarTerm("foo"), StringTerm("bar"), StringTerm("baz")).Value.(Ref)}
+	pkg3 := &Package{Path: RefTerm(VarTerm("foo"), StringTerm("qux"), StringTerm("baz")).Value.(Ref)}
+	assertPackagesEqual(t, pkg1, pkg1)
+	assertPackagesEqual(t, pkg1, pkg2)
+	assertPackagesNotEqual(t, pkg1, pkg3)
+	assertPackagesNotEqual(t, pkg2, pkg3)
+}
+
+func TestPackageString(t *testing.T) {
+	pkg1 := &Package{Path: RefTerm(VarTerm("foo"), StringTerm("bar"), StringTerm("baz")).Value.(Ref)}
+	result1 := pkg1.String()
+	expected1 := "package foo.bar.baz"
+	if result1 != expected1 {
+		t.Errorf("Expected %v but got %v", expected1, result1)
+	}
+}
```

Comment:
These tests for String() seem useful if we expect the property that x.parse().String().parse() = x.parse().  Or more realistically that x.parse().String().parse() = x.parse() up to variable renaming.  Is that what we're going for?  That'd be cool.